src/partial/partial.ts

```diff
@@ -9,11 +9,11 @@ export type PartialDeep<T> = T extends Primitive
   : T extends ReadonlyMap<infer KeyType, infer ValueType>
   ? PartialReadonlyMapDeep<KeyType, ValueType>
   : T extends ReadonlySet<infer ItemType>
-  ? PartialReadonlySetDeep<ItemType> // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  : T extends (...args: any[]) => unknown
-  ? T | undefined
+  ? PartialReadonlySetDeep<ItemType>
   : T extends object
   ? PartialObjectDeep<T>
+  : T extends (...args: any[]) => unknown // eslint-disable-line @typescript-eslint/no-explicit-any
```

Comment:
Considering that a function is always an object, shall we remove this completely?

I mean 

```
: T extends (...args: any[]) => unknown // eslint-disable-line @typescript-eslint/no-explicit-any
? T | undefined
```

Reply:
The only problem with this approach, removing or moving the line up is this scenario.

```ts
type B = () => string;

  const b: PartialDeep<B> = {
    a: 's',
  };
```

This doesn't error. If you instead put back the function check it will error
```
Type '{ a: string; }' is not assignable to type 'B'.   Object literal may only specify known properties, and 'a' does not exist in type 'B'.
```

Reply:
You're right, I'll try to find another way